torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -685,16 +696,51 @@ const ExecutionPlan& ProfilingGraphExecutorImpl::getOptimizedPlanFor(
     return *profiling_plan_;
   }
 
-  auto copy = pr_->graph()->copy();
-  ProfilingRecord::removeProfileCounter(copy->block());
-  runProfilingOptimizations(copy);
-  // replaces a fallback graph inserted by
-  // specialize_autogradzero if one exists
-  replaceFallbackGraphWithFallbackFunction(copy->block());
-  GRAPH_DUMP("Optimized Graph: ", copy);
-  optimized_plan_ =
-      ExecutionPlan(copy, function_name_, *remaining_bailout_depth_);
-  return *optimized_plan_;
+  // deferred just makes compilation lazy but we still compile async
+  auto comp_mode =
```

Comment:
we could completely avoid using `deferred` and maybe save a few cycles on a task construction. It will be more code though.